src/algorithm/ATOMIC.cpp

```diff
@@ -35,6 +35,10 @@ ATOMIC::ATOMIC(const RunParams& params)
 
   setComplexity(Complexity::N);
 
+  setNestedLoops(0);
+  setArrayDimensions(1);
```

Comment:
By Array Dimensions you're basically asking if the problem or the data is 1, 2, 3, etc dimensions? How about calling it ProblemDimensions, Dimensionality, or something like that?
There are some problems where there are arrays with differing dimensionality involved, like in LTIMES there is a 4d loop that goes over 2d and 3d arrays.